src/WebJobs.Extensions/Extensions/Timers/Scheduling/ScheduleMonitor.cs

```diff
@@ -56,32 +56,57 @@ public abstract class ScheduleMonitor
                 lastStatus = new ScheduleStatus
                 {
                     Last = default(DateTime),
-                    Next = nextOccurrence
+                    Next = nextOccurrence,
+                    LastUpdated = now
                 };
                 await UpdateStatusAsync(timerName, lastStatus);
                 recordedNextOccurrence = nextOccurrence;
             }
             else
             {
                 DateTime expectedNextOccurrence;
-                if (lastStatus.Last == default(DateTime))
+
+                // Track the time that was used to create 'expectedNextOccurrence'.
+                DateTime lastUpdated;
+
+                if (lastStatus.LastUpdated != default(DateTime))
                 {
-                    // there have been no executions of the function yet, so compute
-                    // from now
-                    expectedNextOccurrence = schedule.GetNextOccurrence(now);
+                    // If we know when we last calculated the stored 'Next', use the current schedule
+                    // to recalculate. It may have changed, which will be checked below.
+                    expectedNextOccurrence = schedule.GetNextOccurrence(lastStatus.LastUpdated);
+                    lastUpdated = lastStatus.LastUpdated;
                 }
-                else
+                else if (lastStatus.Last != default(DateTime))
                 {
-                    // compute the next occurrence from the last
+                    // Previous versions did not store 'LastUpdated', so the next best alternative is 
+                    // to check 'Last'. 
                     expectedNextOccurrence = schedule.GetNextOccurrence(lastStatus.Last);
+                    lastUpdated = lastStatus.Last;
+                }
+                else
+                {
+                    // If we do not have 'LastUpdated' or 'Last', we don't have enough information to 
+                    // properly calculate 'Next', so we'll calculate it from the current time.
+                    expectedNextOccurrence = schedule.GetNextOccurrence(now);
+                    lastUpdated = now;
                 }
 
                 // ensure that the schedule hasn't been updated since the last
                 // time we checked, and if it has, update the status to use the new schedule
                 if (lastStatus.Next != expectedNextOccurrence)
                 {
+                    // if the schedule has changed and the next occurrence is in the past,
+                    // recalculate it based on the current time as we don't want it to register
+                    // immediately as 'past due'.
+                    if (now > expectedNextOccurrence)
```

Comment:
May want to change `CheckPastDueAsync` comments above to make sure they're consistent with this new delay behavior